File: mycollab-scheduler/src/main/java/com/esofthead/mycollab/schedule/email/format/EmailLinkFieldFormat.java
package com.esofthead.mycollab.schedule.email.format;

import java.lang.reflect.InvocationTargetException;

import org.apache.commons.beanutils.PropertyUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.esofthead.mycollab.schedule.email.MailContext;
import com.hp.gagawa.java.elements.A;
import com.hp.gagawa.java.elements.Span;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.0
 * 
 */
public class EmailLinkFieldFormat extends FieldFormat {
	private static Logger log = LoggerFactory
			.getLogger(EmailLinkFieldFormat.class);

	public EmailLinkFieldFormat(String fieldName, String displayName) {
		super(fieldName, displayName);
	}

	@Override
	public String formatField(MailContext<?> context) {
		Object wrappedBean = context.getWrappedBean();
		Object value;
		try {
			value = PropertyUtils.getProperty(wrappedBean, fieldName);
			return formatEmail((String) value);
		} catch (IllegalAccessException | InvocationTargetException
				| NoSuchMethodException e) {
			log.error("Error", e);
			return new Span().write();
		}
	}

	@Override
	public String formatField(MailContext<?> context, String value) {
		return formatEmail(value);
	}

	private String formatEmail(String value) {
		if (value == null) {
			return new Span().write();
		} else {
			A link = new A();
			link.setStyle("text-decoration: none; color: rgb(36, 127, 211);");
			link.setHref("mailto:" + value.toString());
			link.appendText(value.toString());
			return new Span().appendChild(link).write();
		}
	}

}

File: mycollab-scheduler/src/main/scala/com/mycollab/schedule/email/format/EmailLinkFieldFormat.scala
package com.mycollab.schedule.email.format

import com.hp.gagawa.java.elements.{A, Span}
import com.mycollab.schedule.email.MailContext
import org.apache.commons.beanutils.PropertyUtils
import org.slf4j.LoggerFactory

/**
 * @author MyCollab Ltd.
 * @since 4.6.0
 */
class EmailLinkFieldFormat(fieldName: String, displayName: Enum[_]) extends FieldFormat(fieldName, displayName) {
    val LOG = LoggerFactory.getLogger(classOf[EmailLinkFieldFormat])

    override def formatField(context: MailContext[_]): String = {
        val wrappedBean = context.wrappedBean
        try {
            val value = PropertyUtils.getProperty(wrappedBean, fieldName)
            formatEmail(value.asInstanceOf[String])
        } catch {
            case e: Any =>
                LOG.error("Error", e)
                new Span().write
        }
    }

    override def formatField(context: MailContext[_], value: String): String = formatEmail(value)

    private def formatEmail(value: String): String = {
        if (value == null) {
            new Span().write
        } else {
            val link: A = new A
            link.setStyle("text-decoration: none; color: rgb(36, 127, 211);")
            link.setHref("mailto:" + value.toString)
            link.appendText(value.toString)
            new Span().appendChild(link).write
        }
    }
}
